File: multiple-languages/dotnet/AlibabaCloud.SDK.ROS.CDK.Esa/AlibabaCloud/SDK/ROS/CDK/Esa/IHttpsApplicationConfiguration.cs
using Amazon.JSII.Runtime.Deputy;

#pragma warning disable CS0672,CS0809,CS1591

namespace AlibabaCloud.SDK.ROS.CDK.Esa
{
    /// <summary>Represents a `HttpsApplicationConfiguration`.</summary>
    [JsiiInterface(nativeType: typeof(IHttpsApplicationConfiguration), fullyQualifiedName: "@alicloud/ros-cdk-esa.IHttpsApplicationConfiguration")]
    public interface IHttpsApplicationConfiguration : AlibabaCloud.SDK.ROS.CDK.Core.IResource
    {
        /// <summary>Attribute AltSvc: Function switch, default off.</summary>
        [JsiiProperty(name: "attrAltSvc", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrAltSvc
        {
            get;
        }

        /// <summary>Attribute AltSvcClear: Alt-Svc whether The header contains the clear parameter.</summary>
        /// <remarks>
        /// This parameter is disabled by default.
        /// </remarks>
        [JsiiProperty(name: "attrAltSvcClear", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrAltSvcClear
        {
            get;
        }

        /// <summary>Attribute AltSvcMa: The effective time of the Alt-Svc, in seconds.</summary>
        /// <remarks>
        /// The default value is 86400 seconds.
        /// </remarks>
        [JsiiProperty(name: "attrAltSvcMa", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrAltSvcMa
        {
            get;
        }

        /// <summary>Attribute AltSvcPersist: Alt-Svc whether The header contains the persist parameter.</summary>
        /// <remarks>
        /// This parameter is disabled by default.
        /// </remarks>
        [JsiiProperty(name: "attrAltSvcPersist", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrAltSvcPersist
        {
            get;
        }

        /// <summary>Attribute ConfigId: Config Id.</summary>
        [JsiiProperty(name: "attrConfigId", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrConfigId
        {
            get;
        }

        /// <summary>Attribute ConfigType: The type of the configuration.</summary>
        [JsiiProperty(name: "attrConfigType", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrConfigType
        {
            get;
        }

        /// <summary>Attribute Hsts: Whether to enable HSTS.</summary>
        /// <remarks>
        /// It is disabled by default.
        /// </remarks>
        [JsiiProperty(name: "attrHsts", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrHsts
        {
            get;
        }

        /// <summary>Attribute HstsIncludeSubdomains: Whether to include subdomains in HSTS is disabled by default.</summary>
        [JsiiProperty(name: "attrHstsIncludeSubdomains", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrHstsIncludeSubdomains
        {
            get;
        }

        /// <summary>Attribute HstsMaxAge: The expiration time of HSTS, in seconds.</summary>
        [JsiiProperty(name: "attrHstsMaxAge", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrHstsMaxAge
        {
            get;
        }

        /// <summary>Attribute HstsPreload: Whether to enable HSTS preloading.</summary>
        /// <remarks>
        /// It is disabled by default.
        /// </remarks>
        [JsiiProperty(name: "attrHstsPreload", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrHstsPreload
        {
            get;
        }

        /// <summary>Attribute HttpsForce: Whether to enable forced HTTPS.</summary>
        /// <remarks>
        /// It is disabled by default.
        /// </remarks>
        [JsiiProperty(name: "attrHttpsForce", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrHttpsForce
        {
            get;
        }

        /// <summary>Attribute HttpsForceCode: Forced HTTPS jump status code, value range:.</summary>
        [JsiiProperty(name: "attrHttpsForceCode", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrHttpsForceCode
        {
            get;
        }

        /// <summary>Attribute HttpsNoSniDeny: Whether to enable to reject TLS handshake requests without SNI.</summary>
        /// <remarks>
        /// This parameter is disabled by default.
        /// </remarks>
        [JsiiProperty(name: "attrHttpsNoSniDeny", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrHttpsNoSniDeny
        {
            get;
        }

        /// <summary>Attribute HttpsSniVerify: Whether to enable SNI verification.</summary>
        /// <remarks>
        /// It is disabled by default.
        /// </remarks>
        [JsiiProperty(name: "attrHttpsSniVerify", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrHttpsSniVerify
        {
            get;
        }

        /// <summary>Attribute HttpsSniWhitelist: Specifies the list of allowed SNI whitelists, separated by spaces.</summary>
        [JsiiProperty(name: "attrHttpsSniWhitelist", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrHttpsSniWhitelist
        {
            get;
        }

        /// <summary>Attribute Rule: Rule content, using conditional expressions to match user requests.</summary>
        /// <remarks>
        /// When adding global configuration, this parameter does not need to be set.
        /// </remarks>
        [JsiiProperty(name: "attrRule", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrRule
        {
            get;
        }

        /// <summary>Attribute RuleEnable: Rule switch.</summary>
        /// <remarks>
        /// When adding global configuration, this parameter does not need to be set.
        /// </remarks>
        [JsiiProperty(name: "attrRuleEnable", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrRuleEnable
        {
            get;
        }

        /// <summary>Attribute RuleName: Rule name.</summary>
        /// <remarks>
        /// When adding global configuration, this parameter does not need to be set.
        /// </remarks>
        [JsiiProperty(name: "attrRuleName", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrRuleName
        {
            get;
        }

        /// <summary>Attribute Sequence: Order of rule execution.</summary>
        /// <remarks>
        /// The smaller the value, the higher the priority for execution.
        /// </remarks>
        [JsiiProperty(name: "attrSequence", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrSequence
        {
            get;
        }

        /// <summary>Attribute SiteVersion: The version number of the site configuration.</summary>
        /// <remarks>
        /// For sites that have enabled configuration version management, this parameter can be used to specify the effective version of the configuration site, which defaults to version 0.
        /// </remarks>
        [JsiiProperty(name: "attrSiteVersion", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrSiteVersion
        {
            get;
        }

        [JsiiProperty(name: "props", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-esa.HttpsApplicationConfigurationProps\"}")]
        AlibabaCloud.SDK.ROS.CDK.Esa.IHttpsApplicationConfigurationProps Props
        {
            get;
        }

        /// <summary>Represents a `HttpsApplicationConfiguration`.</summary>
        [JsiiTypeProxy(nativeType: typeof(IHttpsApplicationConfiguration), fullyQualifiedName: "@alicloud/ros-cdk-esa.IHttpsApplicationConfiguration")]
        internal sealed class _Proxy : DeputyBase, AlibabaCloud.SDK.ROS.CDK.Esa.IHttpsApplicationConfiguration
        {
            private _Proxy(ByRefValue reference): base(reference)
            {
            }

            /// <summary>Attribute AltSvc: Function switch, default off.</summary>
            [JsiiProperty(name: "attrAltSvc", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrAltSvc
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute AltSvcClear: Alt-Svc whether The header contains the clear parameter.</summary>
            /// <remarks>
            /// This parameter is disabled by default.
            /// </remarks>
            [JsiiProperty(name: "attrAltSvcClear", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrAltSvcClear
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute AltSvcMa: The effective time of the Alt-Svc, in seconds.</summary>
            /// <remarks>
            /// The default value is 86400 seconds.
            /// </remarks>
            [JsiiProperty(name: "attrAltSvcMa", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrAltSvcMa
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute AltSvcPersist: Alt-Svc whether The header contains the persist parameter.</summary>
            /// <remarks>
            /// This parameter is disabled by default.
            /// </remarks>
            [JsiiProperty(name: "attrAltSvcPersist", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrAltSvcPersist
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute ConfigId: Config Id.</summary>
            [JsiiProperty(name: "attrConfigId", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrConfigId
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute ConfigType: The type of the configuration.</summary>
            [JsiiProperty(name: "attrConfigType", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrConfigType
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute Hsts: Whether to enable HSTS.</summary>
            /// <remarks>
            /// It is disabled by default.
            /// </remarks>
            [JsiiProperty(name: "attrHsts", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrHsts
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute HstsIncludeSubdomains: Whether to include subdomains in HSTS is disabled by default.</summary>
            [JsiiProperty(name: "attrHstsIncludeSubdomains", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrHstsIncludeSubdomains
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute HstsMaxAge: The expiration time of HSTS, in seconds.</summary>
            [JsiiProperty(name: "attrHstsMaxAge", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrHstsMaxAge
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute HstsPreload: Whether to enable HSTS preloading.</summary>
            /// <remarks>
            /// It is disabled by default.
            /// </remarks>
            [JsiiProperty(name: "attrHstsPreload", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrHstsPreload
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute HttpsForce: Whether to enable forced HTTPS.</summary>
            /// <remarks>
            /// It is disabled by default.
            /// </remarks>
            [JsiiProperty(name: "attrHttpsForce", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrHttpsForce
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute HttpsForceCode: Forced HTTPS jump status code, value range:.</summary>
            [JsiiProperty(name: "attrHttpsForceCode", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrHttpsForceCode
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute HttpsNoSniDeny: Whether to enable to reject TLS handshake requests without SNI.</summary>
            /// <remarks>
            /// This parameter is disabled by default.
            /// </remarks>
            [JsiiProperty(name: "attrHttpsNoSniDeny", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrHttpsNoSniDeny
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute HttpsSniVerify: Whether to enable SNI verification.</summary>
            /// <remarks>
            /// It is disabled by default.
            /// </remarks>
            [JsiiProperty(name: "attrHttpsSniVerify", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrHttpsSniVerify
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute HttpsSniWhitelist: Specifies the list of allowed SNI whitelists, separated by spaces.</summary>
            [JsiiProperty(name: "attrHttpsSniWhitelist", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrHttpsSniWhitelist
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute Rule: Rule content, using conditional expressions to match user requests.</summary>
            /// <remarks>
            /// When adding global configuration, this parameter does not need to be set.
            /// </remarks>
            [JsiiProperty(name: "attrRule", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrRule
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute RuleEnable: Rule switch.</summary>
            /// <remarks>
            /// When adding global configuration, this parameter does not need to be set.
            /// </remarks>
            [JsiiProperty(name: "attrRuleEnable", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrRuleEnable
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute RuleName: Rule name.</summary>
            /// <remarks>
            /// When adding global configuration, this parameter does not need to be set.
            /// </remarks>
            [JsiiProperty(name: "attrRuleName", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrRuleName
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute Sequence: Order of rule execution.</summary>
            /// <remarks>
            /// The smaller the value, the higher the priority for execution.
            /// </remarks>
            [JsiiProperty(name: "attrSequence", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrSequence
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute SiteVersion: The version number of the site configuration.</summary>
            /// <remarks>
            /// For sites that have enabled configuration version management, this parameter can be used to specify the effective version of the configuration site, which defaults to version 0.
            /// </remarks>
            [JsiiProperty(name: "attrSiteVersion", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrSiteVersion
            {
                get => GetInstanceProperty<object>()!;
            }

            [JsiiProperty(name: "props", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-esa.HttpsApplicationConfigurationProps\"}")]
            public AlibabaCloud.SDK.ROS.CDK.Esa.IHttpsApplicationConfigurationProps Props
            {
                get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Esa.IHttpsApplicationConfigurationProps>()!;
            }

            /// <summary>The environment this resource belongs to.</summary>
            /// <remarks>
            /// For resources that are created and managed by the CDK
            /// (generally, those created by creating new class instances like Role, Bucket, etc.),
            /// this is always the same as the environment of the stack they belong to;
            /// however, for imported resources
            /// (those obtained from static methods like fromRoleArn, fromBucketName, etc.),
            /// that might be different than the stack they were imported into.
            /// </remarks>
            [JsiiProperty(name: "env", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-core.IResourceEnvironment\"}")]
            public AlibabaCloud.SDK.ROS.CDK.Core.IResourceEnvironment Env
            {
                get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Core.IResourceEnvironment>()!;
            }

            /// <summary>The stack in which this resource is defined.</summary>
            [JsiiProperty(name: "stack", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-core.Stack\"}")]
            public AlibabaCloud.SDK.ROS.CDK.Core.Stack Stack
            {
                get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Core.Stack>()!;
            }

            /// <summary>The construct tree node for this construct.</summary>
            [JsiiProperty(name: "node", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-core.ConstructNode\"}")]
            public AlibabaCloud.SDK.ROS.CDK.Core.ConstructNode Node
            {
                get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Core.ConstructNode>()!;
            }
        }
    }
}

File: multiple-languages/go/alicloudroscdkesa/IHttpsApplicationConfiguration.go
package alicloudroscdkesa

import (
	_jsii_ "github.com/aws/jsii-runtime-go/runtime"

	"github.com/alibabacloud-go/ros-cdk/alicloudroscdkcore"
	"github.com/alibabacloud-go/ros-cdk/alicloudroscdkesa/internal"
)

// Represents a `HttpsApplicationConfiguration`.
type IHttpsApplicationConfiguration interface {
	alicloudroscdkcore.IResource
	// Attribute AltSvc: Function switch, default off.
	AttrAltSvc() interface{}
	// Attribute AltSvcClear: Alt-Svc whether The header contains the clear parameter.
	//
	// This parameter is disabled by default.
	AttrAltSvcClear() interface{}
	// Attribute AltSvcMa: The effective time of the Alt-Svc, in seconds.
	//
	// The default value is 86400 seconds.
	AttrAltSvcMa() interface{}
	// Attribute AltSvcPersist: Alt-Svc whether The header contains the persist parameter.
	//
	// This parameter is disabled by default.
	AttrAltSvcPersist() interface{}
	// Attribute ConfigId: Config Id.
	AttrConfigId() interface{}
	// Attribute ConfigType: The type of the configuration.
	AttrConfigType() interface{}
	// Attribute Hsts: Whether to enable HSTS.
	//
	// It is disabled by default.
	AttrHsts() interface{}
	// Attribute HstsIncludeSubdomains: Whether to include subdomains in HSTS is disabled by default.
	AttrHstsIncludeSubdomains() interface{}
	// Attribute HstsMaxAge: The expiration time of HSTS, in seconds.
	AttrHstsMaxAge() interface{}
	// Attribute HstsPreload: Whether to enable HSTS preloading.
	//
	// It is disabled by default.
	AttrHstsPreload() interface{}
	// Attribute HttpsForce: Whether to enable forced HTTPS.
	//
	// It is disabled by default.
	AttrHttpsForce() interface{}
	// Attribute HttpsForceCode: Forced HTTPS jump status code, value range:.
	AttrHttpsForceCode() interface{}
	// Attribute HttpsNoSniDeny: Whether to enable to reject TLS handshake requests without SNI.
	//
	// This parameter is disabled by default.
	AttrHttpsNoSniDeny() interface{}
	// Attribute HttpsSniVerify: Whether to enable SNI verification.
	//
	// It is disabled by default.
	AttrHttpsSniVerify() interface{}
	// Attribute HttpsSniWhitelist: Specifies the list of allowed SNI whitelists, separated by spaces.
	AttrHttpsSniWhitelist() interface{}
	// Attribute Rule: Rule content, using conditional expressions to match user requests.
	//
	// When adding global configuration, this parameter does not need to be set.
	AttrRule() interface{}
	// Attribute RuleEnable: Rule switch.
	//
	// When adding global configuration, this parameter does not need to be set.
	AttrRuleEnable() interface{}
	// Attribute RuleName: Rule name.
	//
	// When adding global configuration, this parameter does not need to be set.
	AttrRuleName() interface{}
	// Attribute Sequence: Order of rule execution.
	//
	// The smaller the value, the higher the priority for execution.
	AttrSequence() interface{}
	// Attribute SiteVersion: The version number of the site configuration.
	//
	// For sites that have enabled configuration version management, this parameter can be used to specify the effective version of the configuration site, which defaults to version 0.
	AttrSiteVersion() interface{}
	Props() *HttpsApplicationConfigurationProps
}

// The jsii proxy for IHttpsApplicationConfiguration
type jsiiProxy_IHttpsApplicationConfiguration struct {
	internal.Type__alicloudroscdkcoreIResource
}

func (j *jsiiProxy_IHttpsApplicationConfiguration) AttrAltSvc() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrAltSvc",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IHttpsApplicationConfiguration) AttrAltSvcClear() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrAltSvcClear",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IHttpsApplicationConfiguration) AttrAltSvcMa() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrAltSvcMa",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IHttpsApplicationConfiguration) AttrAltSvcPersist() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrAltSvcPersist",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IHttpsApplicationConfiguration) AttrConfigId() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrConfigId",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IHttpsApplicationConfiguration) AttrConfigType() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrConfigType",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IHttpsApplicationConfiguration) AttrHsts() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrHsts",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IHttpsApplicationConfiguration) AttrHstsIncludeSubdomains() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrHstsIncludeSubdomains",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IHttpsApplicationConfiguration) AttrHstsMaxAge() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrHstsMaxAge",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IHttpsApplicationConfiguration) AttrHstsPreload() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrHstsPreload",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IHttpsApplicationConfiguration) AttrHttpsForce() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrHttpsForce",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IHttpsApplicationConfiguration) AttrHttpsForceCode() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrHttpsForceCode",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IHttpsApplicationConfiguration) AttrHttpsNoSniDeny() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrHttpsNoSniDeny",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IHttpsApplicationConfiguration) AttrHttpsSniVerify() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrHttpsSniVerify",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IHttpsApplicationConfiguration) AttrHttpsSniWhitelist() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrHttpsSniWhitelist",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IHttpsApplicationConfiguration) AttrRule() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrRule",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IHttpsApplicationConfiguration) AttrRuleEnable() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrRuleEnable",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IHttpsApplicationConfiguration) AttrRuleName() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrRuleName",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IHttpsApplicationConfiguration) AttrSequence() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrSequence",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IHttpsApplicationConfiguration) AttrSiteVersion() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrSiteVersion",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IHttpsApplicationConfiguration) Props() *HttpsApplicationConfigurationProps {
	var returns *HttpsApplicationConfigurationProps
	_jsii_.Get(
		j,
		"props",
		&returns,
	)
	return returns
}

